Orange/widgets/widget.py

```diff
@@ -197,117 +215,132 @@ def __init__(self, *args, **kwargs):
+                w.setPixmap(QPixmap(filename))
+            setattr(self, attr, w)
+
+    def _create_default_buttons(self):
+        # These button is inserted in buttons_area, if it exists
+        # Otherwise it is up to the widget to add them to some layout
+        if self.graph_name is not None:
+            self.graphButton = QPushButton("&Save Image", autoDefault=False)
+            self.graphButton.clicked.connect(self.save_graph)
+        if hasattr(self, "send_report"):
+            self.report_button = QPushButton("&Report", autoDefault=False)
+            self.report_button.clicked.connect(self.show_report)
+
+    def set_basic_layout(self):
+        self.setLayout(QVBoxLayout())
+        self.layout().setContentsMargins(2, 2, 2, 2)
+        if not self.resizing_enabled:
+            self.layout().setSizeConstraint(QVBoxLayout.SetFixedSize)
 
+        self.want_main_area = self.want_main_area or self.graph_name
+        self._create_default_buttons()
+        self._insert_warning_bar()
+        self._insert_splitter()
         if self.want_control_area:
-            self.controlArea = gui.widgetBox(splitter,
-                                             orientation="vertical",
-                                             margin=0)
-            splitter.setSizes([1])  # Results in smallest size allowed by policy
-
-            if self.graph_name is not None or hasattr(self, "send_report"):
-                leftSide = self.controlArea
-                self.controlArea = gui.widgetBox(leftSide, margin=0)
-            if self.graph_name is not None:
-                self.graphButton = gui.button(leftSide, None, "&Save Graph")
-                self.graphButton.clicked.connect(self.save_graph)
-                self.graphButton.setAutoDefault(0)
-            if hasattr(self, "send_report"):
-                self.report_button = gui.button(leftSide, None, "&Report",
-                                                callback=self.show_report)
-                self.report_button.setAutoDefault(0)
-
-            if self.want_main_area:
-                self.controlArea.setSizePolicy(QSizePolicy.Fixed,
-                                               QSizePolicy.MinimumExpanding)
-            self.controlArea.layout().setContentsMargins(4, 4, 0 if self.want_main_area else 4, 4)
+            self._insert_control_area()
         if self.want_main_area:
-            self.mainArea = gui.widgetBox(splitter,
-                                          orientation="vertical",
-                                          margin=4,
-                                          sizePolicy=QSizePolicy(QSizePolicy.Expanding,
-                                                                 QSizePolicy.Expanding))
-            splitter.setCollapsible(1, False)
-            self.mainArea.layout().setContentsMargins(0 if self.want_control_area else 4, 4, 4, 4)
-
+            self._insert_main_area()
         if self.want_status_bar:
```

Comment:
I suspected this, but I haven't checked. I'll remove it, there are too many whistles and bells here already.  (It will further increase the cyclomatic complexity, though. :)